### DISCUSSION TITLE
Date field updating on save

### DISCUSSION THREAD

**Discussion by kennethsato** [mikro-orm/mikro-orm#4024] [Q&A]
**Describe the bug**
I have an entity that has a MySQL DATE field (not timestamp)
Whenever I try to update any field of my entity, it also updates the DATE field to the previous day.
This started to happen after v4 to v5 Upgrade

example: My entity's date field is 2023-02-08. I update its name field to "Foo". My date field becomes 2023-02-07,

**To Reproduce**
1. Create a entity with a MySQL Date field
2. Update any other field of that entity

**Expected behavior**
Date field should remain unchanged

**Versions**

| Dependency | Version |
| node | 16.17.1 |
| mikro-orm | 5.4.2 |

**Comment by B4nan** [comment]
> | mikro-orm | 5.4.2 |

First, update to latest, this is quite dated. 

> Whenever I try to update any field of my entity, it also updates the DATE field to the previous day.

This sounds like you have some issue with timezones. How do you define the property? How does your ORM config look like? What timezone is your server configured to?

Either way, if you want help with this, you will need to provide more details, ideally a complete reproduction.

**Reply by kennethsato** [reply]
Hello @B4nan 

Actually, I had updated and tested with latest, but got the same problem.
I've mentioned 5.4.2 because it is my production version, but I should have mentioned 5.6.8

I agree that it sounds timezone related.
I define it with Property decorator as a Date type. In JS it has time and it is 00h00m00s.
  @Property()
  data: Date;

And I'm using GMT-3 on my server.

Somewhere, possibly on the when persisting the data to database, there must be some Date/DateTime that is considering these 3 hours shift so it gets back one day.

I think that would be fine if I was persisting that date field. I could do some workaround to add those hours or something.
But what I think is odd is that I'm changing some other field.

**Reply by B4nan** [reply]
But that is a datetime mapping, not date mapping, you need to specify the type. If you want to map it to `Date` object, use `DateType` exported from the core package. Alternatively just map it to string, no need to mess with the timezones here.

**Reply by kennethsato** [reply]
Sorry.
I hadn't seen this:
https://mikro-orm.io/docs/next/custom-types#datetype

I tried to add DateType property like this:
```
  @Property({ type: DateType })
  data: Date;

```
But now how do I query that?
I tried to compare it with a string, but got:

>  Could not convert JS value '2022-9-15' of type 'string' to type DateType

Then I tried to compare it with a Date, but got:

> RangeError: Invalid time value

**Reply by B4nan** [reply]
Why do you map it to the `Date` objects in the first place?

**Reply by B4nan** [reply]
> RangeError: Invalid time value

That's unrelated error, I am sure you can google why is that. You are passing in wrong date format probably.

edit: not just probably, `2022-9-15` is invalid format, it should be `2022-09-15`.

**Reply by kennethsato** [reply]
I map it as Date because I thought that was the "correct" way of doing it.
Would you recommend me to map it as a string?

And you're correct. I misformatted it and it's also not MikroORM related

**Reply by B4nan** [reply]
If it is a date, it's better to map it to `string` I guess. With mapping to `Date` object you open yourself to those kind of timezone issues - because `Date` object is timezone aware, and will default to the current timezone of your system, which may differ from the one your database server is configured to. You could probably get around them with `forceUtcTimezone: true` (or by setting the right timezone in the driver/server), but mapping to `string` makes more sense to me - especially since you want to query it by string anyway.

**Reply by kennethsato** [reply]
Thank you @B4nan .

I agree mapping it to string makes more sense.
I've changed it and it's working.
Still gotta test the whole solution, but I think it'll be fine.

But I still have a question:
When should we use DateType Custom Type?

**Reply by B4nan** [reply]
When you want to map it to the `Date` object. But then you need to use the `Date` object for queries. It is rather an example of how a custom mapped type can look like.
